lib/backend/compat/compat.go

```diff
@@ -646,6 +650,12 @@ func toNodeComponents(d *model.KVPair) (primary *model.KVPair, optional []*model
 			},
 		})
 	}
+	if len(n.OrchRefs) > 0 {
+		optional = append(optional, &model.KVPair{
+			Key:   model.OrchRefKey{Hostname: nk.Hostname},
+			Value: n.OrchRefs,
```

Comment:
Generally, when writing json to the datastore, you want the top level structure to be a map so that it can be extended in the future. I understand that the other things in the node structure are not done like this, so maybe we can get away with it claiming prior art, but ideally instead of storing

```json
[{"nodeName": "mynode", "orchestrator": "k8s"}]
```



we'd store something like

```json
{
  "orchRefs": [{"nodeName": "mynode", "orchestrator": "k8s"}]
}
```

That does make the code a bit ickier. @robbrockbank do you have a preference? I'm tempted to say just store a list given this will change in the v2 datamodel + etcdv3 and it matches what we do for other node subcomponents.

Reply:
Gah! I had originally done that but thought it felt clunky in the struct.

Reply:
I haven't added this on my latest commit, but if we prefer it I am happy to oblige.

To verify what we're looking for, we would want to contain the slice of `OrchRefs` in a struct, which would mean getting that slice would be `Node.OrchRefs.OrchRefs`? Or maybe something similar but with less redundant naming.